### PR TITLE
BZ-1125077 - Limiting serialization classes in a deployment

### PR DESCRIPTION
**Pull request by mrietveld**
Depends on: 

- https://github.com/droolsjbpm/droolsjbpm-knowledge/pull/109

Depended on by:

- https://github.com/droolsjbpm/droolsjbpm-integration/pull/248
- https://github.com/droolsjbpm/kie-docs/pull/22

### CODE REVIEW

File: jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/DeploymentServiceTest.java [hunk 1]
```diff
@@ -365,4 +364,10 @@ public void testDeploymentWithDeclaredTypeInDRL() {
 		}
 
     }
+
+    @Test
+    public void testDeploymentWithSerializationClassesLimited() {
```
Comment (mswiderski) [inline jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/DeploymentServiceTest.java]: isn't this test missing something?

File: jbpm-runtime-manager/src/test/java/org/jbpm/runtime/manager/impl/deploy/DeploymentDescriptorMergerTest.java [hunk 2]
```diff
@@ -494,13 +516,35 @@ public void testDeploymentDesciptorMergeMergeCollectionsAvoidDuplicatesNamedObje
...
+
+	    Matcher matcher = jarLocRegex.matcher(codeLocStr);
+	    assertTrue( "Fix regular expression: " + jarLocRegexStr, matcher.matches() );
+	    double jarVersion = Double.parseDouble(matcher.group(1));
+
+	    DeploymentDescriptorImpl depDesc = new DeploymentDescriptorImpl();
+
+	    assertTrue( "The default value of 'limitSerializationClasses is FALSE in 6.x and TRUE in 7.x",
+	            jarVersion < 7.0d && ! depDesc.getLimitSerializationClasses()
+	            || jarVersion >= 7.0d && depDesc.getLimitSerializationClasses() );
```
Comment (mswiderski) [inline jbpm-runtime-manager/src/test/java/org/jbpm/runtime/manager/impl/deploy/DeploymentDescriptorMergerTest.java]: is there anywhere logic that actually changes the value based on version of jBPM code base?

**Comment by mswiderski** [comment]
@mrietveld looks like there are test failures related to this changes:
```
java.lang.RuntimeException: java.lang.NullPointerException
	at org.jbpm.kie.services.impl.KModuleDeploymentService.addClassToDeployedUnit(KModuleDeploymentService.java:402)
	at org.jbpm.kie.services.impl.KModuleDeploymentService.processResources(KModuleDeploymentService.java:394)
	at org.jbpm.kie.services.impl.KModuleDeploymentService.deploy(KModuleDeploymentService.java:165)
	at org.jbpm.kie.services.test.ClassloaderKModuleDeploymentServiceTest.testDeploymentOfProcesses(ClassloaderKModuleDeploymentServiceTest.java:94)
```

File: jbpm-runtime-manager/src/test/java/org/jbpm/runtime/manager/impl/deploy/DeploymentDescriptorMergerTest.java (hunk 2 above)
Comment (mrietveld) [inline jbpm-runtime-manager/src/test/java/org/jbpm/runtime/manager/impl/deploy/DeploymentDescriptorMergerTest.java]: Github is having problems displaying the actual code in the PR: see https://github.com/droolsjbpm/jbpm/pull/356/files#diff-d4f4a2a33edb3e2f281c94f1dfa8d522R539

**Comment by mrietveld** [comment]
NPE fixed here: https://github.com/droolsjbpm/jbpm/pull/356/files#diff-545cfd614f196932295be33d02842c55R404

**Pull request closed without merging**